model/src/test/kotlin/utils/ExtensionsTest.kt

```diff
@@ -0,0 +1,188 @@
+                result shouldContain packageA1
+                result shouldContain packageA2
+            }
+
+            "Apply license choices" {
+                val packageA = Package.EMPTY.copy(
+                    id = Identifier("Maven:org.ossreviewtoolkit:a:1.0.0"),
+                    declaredLicenses = sortedSetOf("license-a", "license-c"),
+                    declaredLicensesProcessed = ProcessedDeclaredLicense("license-a OR license-c".toSpdx())
+                )
+                val packageB = Package.EMPTY.copy(
+                    id = Identifier("Maven:org.ossreviewtoolkit:b:0.42"),
+                    declaredLicenses = sortedSetOf("license-b"),
+                    declaredLicensesProcessed = ProcessedDeclaredLicense("license-b".toSpdx())
+                )
+                val packageC = Package.EMPTY.copy(
+                    id = Identifier("Maven:org.ossreviewtoolkit:c:1.2.3"),
+                    declaredLicenses = sortedSetOf("license-c"),
+                    declaredLicensesProcessed = ProcessedDeclaredLicense("license-c".toSpdx())
+                )
+                val packages = listOf(packageA, packageB, packageC)
+
+                val licenseCategorizations = listOf(
+                    LicenseCategorization("license-a".toSpdx() as SpdxSingleLicenseExpression, setOf("category-a")),
+                    LicenseCategorization(
+                        "license-ab".toSpdx() as SpdxSingleLicenseExpression,
+                        setOf("category-a", "category-b")
+                    ),
+                    LicenseCategorization("license-b".toSpdx() as SpdxSingleLicenseExpression, setOf("category-b")),
+                    LicenseCategorization("license-c".toSpdx() as SpdxSingleLicenseExpression, setOf("category-c")),
+                )
+
+                val categories = listOf("category-a", "category-b")
+                val result = packages.filterByLicenseCategories(
+                    categories,
+                    licenseCategorizations,
+                    createOrtResultWithPackages(packages)
+                )
+
+                result shouldHaveSize 2
+                result shouldContain packageA
+                result shouldContain packageB
+                result shouldNotContain packageC
+            }
+        }
+    }
+
+    private fun createOrtResultWithPackages(packages: List<Package>) =
+        OrtResult.EMPTY.copy(
+            analyzer = AnalyzerRun(
+                environment = Environment(),
+                config = AnalyzerConfiguration(),
+                result = AnalyzerResult.EMPTY.copy(
+                    packages = packages.map { it.toCuratedPackage() }.toSortedSet()
+                )
+            ),
+            repository = Repository.EMPTY.copy(
+                config = RepositoryConfiguration(
+                    licenseChoices = LicenseChoices(
+                        packageLicenseChoices = listOf(
+                            PackageLicenseChoice(
```

Comment:
I think this should be a parameter of the function because it's specific to one test case.

Reply:
Moved the `LicenseChoices` to the parameters.